x-pack/plugins/uptime/server/rest_api/synthetics_service/get_service_locations.ts

```diff
@@ -13,6 +13,11 @@ export const getServiceLocationsRoute: UMRestApiRouteFactory = () => ({
   method: 'GET',
   path: API_URLS.SERVICE_LOCATIONS,
   validate: {},
-  handler: async ({ server }): Promise<any> =>
-    getServiceLocations({ manifestUrl: server.config.service!.manifestUrl! }),
+  handler: async ({ server }): Promise<any> => {
+    if (server.syntheticsService.locations.length > 0) {
+      return { locations: server.syntheticsService.locations };
+    }
+
+    return getServiceLocations({ manifestUrl: server.config.service!.manifestUrl! });
+  },
```

Comment:
I'm not sure about this logic. I see that fetching locations was moved to the synthetics service init step. This step is triggered on plugin start, which is triggered when Kibana started. If this process is successful, line 17 will always remain true. This means that there will only be 1 fetch of locations for the entire life of the Kibana instance, until it is restarted. Is this desired? It is unclear to me what our rollout of new locations will look like, but in this scenario, new users would never get access to new locations that may have been rolled out, unless they forced a restart of Kibana or upgraded their stack (which forces a restart of Kibana).